src/renderers/dom/shared/DOMProperty.js

```diff
@@ -226,6 +237,39 @@ var DOMProperty = {
     return false;
   },
 
+  /**
+   * Checks whether a property name is a writeable attribute.
+   * @method
+   */
+  isWriteable: function(name, value) {
+    if (DOMProperty.isReservedProp(name)) {
+      return false;
+    }
+
+    if (value == null || DOMProperty.properties.hasOwnProperty(name)) {
+      return true;
+    }
+
+    if (DOMProperty.isCustomAttribute(name)) {
+      return true;
+    }
+
+    return typeof value === 'string';
+  },
```

Comment:
@gaearon Just to make sure I interpreted your comment correctly, custom attribute logic follows:

1. Reserved words are not allowed
2. null values are allowed for property removal
3. Attributes we know about are allowed
4. Data and ARIA attributes are allowed
5. For anything else, if the value passed for the attribute is a string let it be assigned

A nice side-effect of this is that it gets around needing to know about event registration names for server side rendering. However I would love to allow numbers, strings, and boolean values for this check. If we did that, we could remove the `isCustomAttribute` check all together, cutting a bit of code and eliminating the overhead of the regex matching for every attribute assignment. Maybe that has to happen later.